### PR TITLE
NEVISACCESSAPP-5921: Adapted PR workflows to work with private repositories.

### PR DESCRIPTION
**Pull request by viktor-gulyas**

### CODE REVIEW

**Review by daniel-toth-leeder — APPROVED** [review]

**Review by daniel-toth-leeder — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -60,10 +82,24 @@ jobs:
...
         if: ${{ inputs.build-ios }}
-        run: flutter build ios --no-codesign
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ env.RUBY_VERSION }}
+          bundler-cache: true
+          working-directory: 'ios'
+
+      # Build the iOS app
+      - name: Run native dependency preparation
```
Comment (tamas-toth) [inline .github/workflows/build.yml:95]: 🔧  I think this name is just a leftover, it can be `Build iOS App`

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -77,10 +113,23 @@ jobs:
...
+      - name: Cache RubyGem Dependencies
+        if: ${{ inputs.build-android }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ env.RUBY_VERSION }}
+          bundler-cache: true
+          working-directory: 'android'
+
+      # Build the Android app
+      - name: Run native dependency preparation
```
Comment (tamas-toth) [inline .github/workflows/build.yml:126]: 🔧  I think this name is just a leftover, it can be `Build Android App`

File: ios/Podfile [hunk 3]
```diff
@@ -7,8 +7,8 @@ platform :ios, deployment_target
 ENV['COCOAPODS_DISABLE_STATS'] = 'true'
 
 project 'Runner', {
-'Debug' => :debug,
-	'Profile' => :release,
+	'Debug' => :debug,
```
Comment (tamas-toth) [inline ios/Podfile:10]: ❓ Are these changes just leftovers?

Comment (viktor-gulyas) [reply ios/Podfile:10]: Nope, I think `profile` should be executed with debug flavour.

**Review by tamas-toth — APPROVED** [review]

**Pull request merged**